src/tcp/socks5.rs

```diff
@@ -330,6 +371,92 @@ where
         self.len = 2;
     }
 
+    async fn prepare_send_gssapi_subnego_token(&mut self) -> Result<()> {
+        if let Authentication::Gssapi { gssapi_authenticator } = &self.auth {
+        /*
+            https://www.rfc-editor.org/rfc/rfc1961
+            
+            The security context protection level is sent from client to server
+            and vice versa using the following protected message format:
+
+            +------+------+------+.......................+
+            + ver  | mtyp | len  |   token               |
+            +------+------+------+.......................+
+            + 0x01 | 0x02 | 0x02 | up to 2^16 - 1 octets |
+            +------+------+------+.......................+
+
+            Where:
+                - "ver" is the protocol version number, here 1 to represent the first version of the SOCKS/GSS-API protocol
+
+                - "mtyp" is the message type, here 2 to represent a protection-level negotiation message
+
+                - "len" is the length of the "token" field in octets
+
+                - "token" is the GSS-API encapsulated protection level
+        */
+            
+            self.ptr = 0;
+            self.buf[0] = 0x01;  // ver
+            self.buf[1] = 0x02;  // mtyp
+            let snego_token = gssapi_authenticator.gssapi_authenticator.get_protection_level().await?;
+            let snego_token_bytes = snego_token.as_bytes();
+            let snego_token_len = snego_token_bytes.len();
+            self.buf[2] = snego_token_len as u8; // len
```

Comment:
It looks inconsistent with the spec. The spec says `len` has two bytes.

Reply:
And because `token` can have up to 65535 bytes, it doesn't seem a good idea to put GSS-API auth messages in `buf`.

Reply:
Yep, seems like a big miss, will address these, will create a vec<u8> for this buf separately and return it to be written.